app/controllers/spotlight/catalog_controller.rb

```diff
@@ -39,7 +39,15 @@ class CatalogController < ::CatalogController
 
     def show
       super
-
+      @docs = []
+      if  @document.sidecars.length>0 && @document.sidecars.first.compound?
```

Comment:
It seems like you're repeating this logic both here and in `edit`.  Maybe an accessor on the `SolrDocumentSidecar` model would help?